typescript/cli/tests/warp-apply.zx-test.ts

```diff
@@ -0,0 +1,52 @@
+import { Wallet, providers } from 'ethers';
+
+import { HypNative__factory } from '@hyperlane-xyz/core';
+import { assert, eqAddress } from '@hyperlane-xyz/utils';
+
+import { getDeployedWarpAddress, updateWarpOwner } from './commands/helpers.js';
+import {
+  hyperlaneCoreDeploy,
+  hyperlaneWarpApply,
+  hyperlaneWarpDeploy,
+} from './commands/warp.js';
+
+/// To run: 1) start an anvil, 2) yarn run tsx tests/warp.zs-test.ts inside of cli/
```

Comment:
This a more of a PoC. Will clean up & integrate this more (CI, use chai/expect, etc.) for e2e testing work